boris-client/src/Boris/Client/Log.hs

```diff
@@ -21,8 +23,17 @@ import           P
 
 import           System.IO (IO)
 
+import           Snooze.Balance.Control (BalanceConfig)
+
 import           Twine.Snooze (seconds)
 
+import           X.Control.Monad.Trans.Either (EitherT)
+
+log :: BalanceConfig -> BuildId -> EitherT BorisHttpClientError IO (Maybe LogData)
```

Comment:
This type doesn't look right to me which might be throwing things off, mainly the LogData type. Will make a more detailed comment below.